core/src/main/java/io/grpc/Status.java

```diff
@@ -328,9 +328,9 @@ public static Status fromCodeValue(int codeValue) {
   public static Status fromThrowable(Throwable t) {
     for (Throwable cause : Throwables.getCausalChain(t)) {
       if (cause instanceof StatusException) {
-        return ((StatusException) cause).getStatus();
+        return ((StatusException) cause).getStatus().withCause(t);
```

Comment:
Discussed offline. It seems this doesn't get us what we want. If there are problems in the transport, we should consider them on a case-by-case basis.